File: BasicHttpServer/Models/HttpRequest.py
from typing import Dict

class HTTPRequest:
   
    def __init__(self) -> None:
        self.route: str = ''
        self.method: str = ''
        self.version: str = ''
        self.headers: Dict[str, str] =  {}
        self.body: bytes = None
        
    def appendHeader(self, header: Dict[str, str]):
        keys =list(header.keys())
        if header is not None and keys[0] not in self.headers:
            self.headers[keys[0]] = header[keys[0]]
        

File: rustHttpStaticServer/src/http_model/http_request.rs
use std::collections::HashMap;

#[path = "./http_method.rs"]
mod http_method;

#[derive(Debug)]
pub struct HttpRequest {
    pub(crate) method: http_method::HttpMethod,
    pub(crate) version: String,
    pub(crate) route: String,
    pub(crate) headers: HashMap<String, String>,
    pub(crate) body: Vec<u8>,
}